apps/ios/LiveVideo/LiveVideo/Twilio/Stream/StreamViewModel.swift

```diff
@@ -106,11 +107,32 @@ class StreamViewModel: ObservableObject {
     private func handleError(_ error: Error) {
         streamManager.disconnect()
         
-        if let error = error as? LiveVideoError, error.isStreamEndedByHostError {
+        if error.isStreamEndedByHostError {
             alertIdentifier = .streamEndedByHost
+        } else if error.isSpeakerMovedToViewersByHostError {
+            alertIdentifier = .speakerMovedToViewersByHost
+            streamManager.changeRole(to: .viewer)
         } else {
             self.error = error
             alertIdentifier = .error
         }
     }
 }
+
+private extension Error {
+    var isSpeakerMovedToViewersByHostError: Bool {
+        if case .speakerMovedToViewersByHost = self as? LiveVideoError {
+            return true
+        }
+
+        return false
+    }
+
+    var isStreamEndedByHostError: Bool {
+        if case .streamEndedByHost = self as? LiveVideoError {
+            return true
+        }
+
+        return false
+    }
+}
```

Comment:
These are not errors, they are reflection of  admin actions. IMHO

Reply:
Yeah I get how one can interpret it that way.

I think it is a gray area. These are fatal errors from `RoomManager` in the sense that the user has been unexpectedly disconnected from the room. And also these cases are derived by errors received from the video SDK. I'm just transforming them a little for this app.

I guess the best alternative would be to create 2 new publishers instead of reusing `roomDisconnectPublisher`. But now there are multiple "disconnect" publishers which seems more complex and problematic. Or instead of `roomDisconnectPublisher` sending an `Error` I could create a new type that can hold an error or other information. But that seems unconventional and will also complicate usage.

If these states were received from `Sync` instead of the video SDK and "events" were sent instead of errors I think we would not use errors for this. But for a number of reasons we are receiving this state from the video SDK as errors.

So I totally get your perspective and it's good feedback. But I think all things considered using errors is the best solution because it is much simpler, the state is derived from video SDK errors, and I wouldn't go so far as to say using errors for these events is incorrect or confusing.